include/zephyr/drivers/adc.h

```diff
@@ -644,6 +606,19 @@ static inline int z_impl_adc_channel_setup(const struct device *dev,
 	return api->channel_setup(dev, channel_cfg);
 }
 
+/**
+ * @brief Configure an ADC channel from a struct adc_dt_spec.
+ *
+ * @param spec ADC specification from Devicetree.
+ *
+ * @return A value from adc_channel_setup().
+ * @see adc_channel_setup()
+ */
+static inline int adc_channel_setup_dt(const struct adc_dt_spec *spec)
+{
+	return adc_channel_setup(spec->dev, &spec->channel_cfg);
```

Comment:
`spec->channel_cfg_dt_node_exists` should be checked before this call is done.

Reply:
done, and documented return code